Renaming CanTherm as Arkane

alongd opened the pull request:
The name CanTherm no longer represents this software, and we have agreed to rename it as:
`Arkane - Automated Reaction Kinetics and Network Exploration`

This PR refactors CanTherm into Arkane.
Backwards compatibility of input files is conserved, the executable is now `arkane.py` instead of `cantherm.py`.

Examples and documentation were changed accordingly.

(We should also change the labelling of PRs and issues)

alongd commented:
I agree that `Arkane` looks better than `ARKANE`, I changed all occurrences (I originally followed the capitalization style of `RMG`, but it doesn't look as good when the word is pronounced with its vowels).

Sure, how do we make it a root level module?

mliu49 commented:
I think you would just move the entire arkane directory up one level to RMG-Py/arkane and update the imports. It might be more trouble than it's worth though, so I think you can decide whether or not you want to try.

alongd commented:
I'll skip making it a root level module for now, I'll consider it later on.
I think this PR is ready to merge, let me know if you have any further comments

mliu49 commented on documentation/source/conf.py:
```diff
@@ -219,7 +219,7 @@
 # (source start file, target name, title, author, documentclass [howto/manual]).
 latex_documents = [
-  ('latex-rmg', 'RMG-Py_and_CanTherm_Documentation.tex', u'RMG-Py and CanTherm Documentation',
+  ('latex-rmg', 'RMG-Py_and_ARKANE_Documentation.tex', u'RMG-Py and Arkane Documentation',
```
Change capitalization of Arkane here?

mliu49 commented on documentation/source/latex-rmg.rst:
```diff
@@ -11,21 +11,21 @@ steps using a general understanding of how molecules react. This documentation i
-+---------------------------------------------+----------------------------------------------------------------------------------------+
 | create mechanisms automatically             | Download RMG with the RMG User's Guide                                                 |
 +---------------------------------------------+----------------------------------------------------------------------------------------+
+| make transition state theory calculations   | Run Arkane after downloading RMG. See Arkane's User's Guide                            |
```
Minor, but perhaps `See the Arkane User's Guide` instead to be consistent?

mliu49 commented on examples/arkane/networks/methoxy/input.py:
```diff
@@ -2,11 +2,12 @@
+This example illustrates how to manually set up an Arkane input file for a small P-dep reaction system (using only the
+RRHO assumption, and without tunneling, although this can be easily implemented). Such a calculation is desirable if
+the user wishes to supply experimentally determined frequencies, for example. Although some commented notes below may
+be useful, see http://greengroup.github.io/RMG-Py/users/arkane/index.html for more documented information about
```
I think this URL should also be updated: http://reactionmechanismgenerator.github.io/RMG-Py/users/cantherm/index.html

mliu49 commented on rmgpy/arkane/main.py:
```diff
@@ -225,7 +222,7 @@ def execute(self):
         # Initialize the logging system (both to the console and to a file in the
         # output directory)
-        self.initializeLog(self.verbose, os.path.join(self.outputDirectory, 'cantherm.log'))
+        self.initializeLog(self.verbose, os.path.join(self.outputDirectory, 'Arkane.log'))
```
I would prefer the file name to be completely lowercase, but it's your choice. The gitignore file should also be updated if you decide to change it.

alongd replied:
Done. Why should `.gitignore` be changed?
Isn't
```python
# Example output files
examples/*
```
enough?

mliu49 commented on rmgpy/arkane/pdepTest.py:
```diff
@@ -43,31 +43,31 @@
 
 @attr('functional')
-class CanthermTest(unittest.TestCase):
+class ARKANETest(unittest.TestCase):
```
Rename class to `ArkaneTest`?

mliu49 commented:
Also, was the HSO2=HSOO example intentionally added in this PR?

alongd commented:
Thanks @mliu49 
I addressed all comments, and also made Arkane a root-level module. So far it seems to work smoothly

alongd commented:
All of Arkane's examples run successfully on this branch

alongd commented:
@mliu49, this Arkane renaming branch is now rebased and squashed. I believe I addressed all comments, and I also ran all of Arkane's examples on this branch successfully. This PR should be good to go.

alongd commented:
We should also think about either changing or adding a GitHub label for Arkane

mliu49 commented on examples/arkane/species/C2H4/ethene.log:
```diff
@@ -1826,4 +1826,4 @@
  Job cpu time:  0 days  0 hours  0 minutes 39.5 seconds.
  File lengths (MBytes):  RWF=    329 Int=      0 D2E=      0 Chk=     11 Scr=      1
- Normal termination of Gaussian 03 at Fri Feb  9 00:59:20 2007.
+Normal termination of Gaussian 03 at Fri Feb 9 00:59:20 2007.
```
Were the spacing changes in this file intentional?

alongd replied:
No. I might have copy-pasted this file, so perhaps the text editor added them. I'll revert it

alongd commented on setup.py:
```diff
@@ -208,7 +211,7 @@ def getCanthermExtensionModules():
         if file.endswith('.py') or file.endswith('.pyx'):
             if 'Test' not in file and '__init__' not in file:
-                if not root.endswith('rmgpy/cantherm/files'):
+                if not root.endswith('rmgpy/arkane/files'):
```
You're right, something's strange here. The loop in line 208 walks through rmgpy, and seems like setting the module on line 215 should also be changed (it's no longer in rmgpy). I'm not an expert on building/installing python modules, but is it at all needed if arkane is recognized now at the root level?

alongd commented:
If #1502 is ready, please go ahead with merging it.

alongd commented:
@mliu49, all modifications were done as we discussed, let me know how the binaries test went

mliu49 commented:
I could compile the binary without any problems and was able to successfully import the arkane module.

Last question, have you compiled the documentation locally to check that it looks fine?

alongd commented:
Yes, I recompiled and fixed the documentation where it didn't look good

alongd commented:
Haha :)
Well, @xiaoruiDong already called it Arkane in his talk today... might as well make it so
